packages/actor-abstract-path/lib/ActorAbstractPath.ts

```diff
@@ -131,10 +168,19 @@ export abstract class ActorAbstractPath extends ActorQueryOperationTypedMediated
     if (termString in termHashesGlobal) {
       const objects = await termHashesGlobal[termString];
       for (const object of objects) {
-        if (!termHashesCurrentSubject[termToString(object)]) {
-          termHashesCurrentSubject[termToString(object)] = true;
-          (<any> it)._push(Bindings({ [subjectString]: subjectVal, [objectString]: object }));
-        }
+        await this.ALPTwoVariables(
```

Comment:
This seems like a drastic change compared to what was here originally. Was there something wrong previously?

Reply:
Yes, there was. We only added the results of the mediate from one step to the array, and then we just pushed those. But we have to look further and add their neighbours and so on. But still will there be no work done twice, because the neighbour of the neighbour will already have been calculated as well, so they will have to check that array as well, but not mediate for the same subject more then once.

Reply:
I found this mistake: 
:a : p : c
:c : p : d
:d : p : e
: b : p : c
with a query as ?x :p* ?y : 
:b :c en :b :d where found (as c is common so one step further d is added as waiting on the array in the termhashes)
:b :e was not found, because it is 2 steps from the common c.